### DISCUSSION TITLE
How to set variations when importing as maya data with python

### DISCUSSION THREAD

**Discussion by jhultgre** [Autodesk/maya-usd#2932] [Q&A]
Is there a way to select which variant sets I want loaded when importing a usd file as maya data using mayaUSDImport command?

I'm going to have a variant set called DetailLevel on hundreds of assets and I need to be able to change it on all of them when importing the scene.

When I load it as a proxy shape I can traverse the stage and set it, but how do I do that for maya data?

```
        # load style
        if self.import_style_radios.checkedButton().text() == 'Import as USD Stage':
            import mayaUsd
            shapenode = pm.createNode('mayaUsdProxyShape', name=file_name.split('.')[0] + 'Shape')
            shapenode.filePath.set(usd_local)
            transform = shapenode.getParent()
            transform.rename(file_name.split('.')[0])

            stage = mayaUsd.lib.GetPrim(str(shapenode)).GetStage()

            # TODO: add ui to select
            for prim in stage.TraverseAll():
                vset = prim.GetVariantSets()
                if vset.HasVariantSet('DetailLevel'):
                    vset.GetVariantSet('DetailLevel').SetVariantSelection('proxy')

        else:
            kwargs = {}
            if self.deinstance_cb.isChecked():
                # uninstance
                kwargs['importInstances'] = False
            if self.shader_cb.isChecked():
                # replace shaders
                kwargs['shadingMode'] = [['none', 'default']]
                
            # How to set variants?
            pm.mayaUSDImport(file=usd_local, **kwargs)
```

**Comment by jhultgre** [comment]
Found I can use this to set the variants.
It would really be great if there were examples in the doc instead of having to use the terrible github search to find a test file that does what you are looking for.

`kwargs['variant'] =[('DetailLevel', 'proxy')]`

**Reply by daviddeacon** [reply]
+1 would love more python example snippets.

**Comment by santosd** [comment]
@jhultgre thank you for posting your findings. I agree that more Python examples would be great to have. It is something we have discussed internally and hope to spend more time on doing.
